front-end/src/Pages/CreateRoom.tsx

```diff
@@ -0,0 +1,107 @@
+import styles from './CreateRoom.module.css'
+import Container from '../components/UI/Container'
+import loggedInContext from '../utils/Contexts/loggedInContext'
+import { useContext, useEffect, useRef } from 'react'
+import Button from '../components/UI/Button'
+import axios from 'axios'
+import notificationContext from '../utils/Contexts/notificationContext'
+import { withRouter } from 'react-router'
+import { generateRandomRoom } from '../utils/Rooms'
+import { useState } from 'react'
+import BottomFormPopup from '../components/UI/ButtonFormPopup'
+import Heading from '../components/UI/Heading'
+import InputField from '../components/UI/InputField'
+import Authenticated from '../components/Other/Authenticated'
+import { Link } from 'react-router-dom'
+export interface ICreateRoomProps {
+  history: any
+}
+
+function CreateRoom(props: ICreateRoomProps) {
+  const { isLoggedIn } = useContext(loggedInContext)
+  const roomNameRef = useRef<HTMLInputElement>(null)
+  const notifCtx = useContext(notificationContext)
+  const [roomId, setRoomId] = useState('')
+  const [errorShow, setErrorShow] = useState<boolean>(false)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  useEffect(() => {
+    const getRoomId = async () => {
+      const id = await generateRandomRoom()
+      setRoomId(id)
+    }
+    if (isLoggedIn) getRoomId()
+  }, [isLoggedIn])
+
+  const createRoom = async (e: any) => {
+    if (roomNameRef.current && roomNameRef.current.value.length !== 0) {
+      const result = await axios.post(
+        'http://localhost:3000/rooms/add',
+        {
+          roomId,
```

Comment:
Actually, roomId is generated by the server in case of creating a room and then returned in the result. So, we probably need to change some functionality here.

Reply:
Yup you're right. Have fixed this now